Katello/smart_proxy_container_gateway
# Fixes #31484 - Add test infrastructure

Introduces rubocop, github actions, and unit tests.

To test/review:
1) Checkout this PR and try `bundle install`, `bundle exec rake test`, and `bundle exec rubocop`.
2) Look over the tests and see if they make sense
3) Check over the GitHub action and the RuboCop rules.
